src/msw/overlay.cpp

```diff
@@ -100,12 +125,21 @@ void wxOverlayImpl::Init(wxDC* dc, int , int , int , int )
 
     m_window = dc->GetWindow();
 
-    m_rect.SetSize(m_window->GetClientSize());
-    m_rect.SetPosition(m_window->GetScreenPosition());
+    // The rectangle is in screen coordinates
+    m_rect = dc->GetWindowRect();
 
-    m_bitmap.CreateWithLogicalSize(m_rect.GetSize(), m_window->GetDPIScaleFactor());
+    if ( IsUsingConstantOpacity() )
+    {
+        m_bitmap.CreateWithDIPSize(m_rect.GetSize(), m_window->GetDPIScaleFactor());
+    }
+    else
+    {
+        m_bitmap.CreateWithDIPSize(m_rect.GetSize(), m_window->GetDPIScaleFactor(), 32);
+        m_bitmap.UseAlpha();
+    }
 
-    m_overlayWindow = wxCreateOverlayWindow(m_rect);
+    HWND hWndInsertAfter = ::GetNextWindow(GetHwndOf(m_window), GW_HWNDPREV);
```

Comment:
This change is mainly to prevent the overlay window from being placed on top of/covering any floating window.

**Before:**

![overlay_bad](https://github.com/wxWidgets/wxWidgets/assets/7704771/8e01671b-9f6e-4bc6-aabf-49f968f5b5a3)

**After:**

![overlay_good](https://github.com/wxWidgets/wxWidgets/assets/7704771/2bd98289-8629-4cbc-aace-7fe1f6a5ea05)

Unfortunately, this breaks rubberbanding in the drawing sample. That is, the rubberband no longer appears there! What actually solves both problems is doing the following:
```
    const auto win = wxGetTopLevelParent(m_window);
    HWND hWndInsertAfter = ::GetNextWindow(GetHwndOf(win), GW_HWNDPREV);
    m_overlayWindow = wxCreateOverlayWindow(m_rect, m_alpha, hWndInsertAfter);
```
Which I'll commit if you don't see any problem in doing so (of course) ?